AIO/src/org/aio/activities/quests/CooksAssistant.java

```diff
@@ -0,0 +1,257 @@
+
+    	// Get pot
+		if(!getInventory().contains("Pot") && !getInventory().contains("Pot of flour") && !getInventory().contains("Bucket of milk")) {
+			
+			getGroundItem(cookRoom, "Pot");
+
+		}
+		
+		// Get bucket
+		else if(!getInventory().contains("Bucket") && !getInventory().contains("Bucket of milk")) {
+			
+			getGroundItem(basement, "Bucket");
+
+		}
+    	
+		// Milk cow
+		else if(getInventory().contains("Bucket") && !getInventory().contains("Bucket of milk")) {
+			
+			getItemFromObject(cow, "Bucket of milk", "Dairy cow", "Milk");
+
+		}
+		
+		// Get egg
+		else if(!getInventory().contains("Egg") && getInventory().contains("Bucket of milk")) {
+			
+			getGroundItem(chicken, "Egg");
+			
+    	}
+    	
+    	// Get flour
+		else if(!getInventory().contains("Pot of flour") && getInventory().contains("Egg") && getInventory().contains("Bucket of milk")) {
+
+			// Get grain
+			if(!put && !getInventory().contains("Grain"))
+				getItemFromObject(wheat, "Grain", "Wheat", "Pick");
+			
+			// Put grain
+			if(!put && !operated && getInventory().contains("Grain")) {
+				
+				if(upper.contains(myPlayer())){
+					
+					if(getInventory().getSelectedItemName() != null && getInventory().getSelectedItemName().equals("Grain")) {
+						
+						RS2Object hopper = getObjects().closest(n -> n.getName().equals("Hopper"));
+							
+						if(hopper != null && hopper.interact("Use"))
+							Sleep.sleepUntil(() -> getChatbox().contains(MessageType.GAME, "There is already grain in the hopper.") | (!myPlayer().isAnimating() && !getInventory().contains("Grain")), 15000);
+						if(getChatbox().contains(MessageType.GAME, "There is already grain in the hopper.") | !getInventory().contains("Grain"))
+							put = true;
+						
+					} else {
+						
+						if(getInventory().interact("Use", "Grain"))
+							Sleep.sleepUntil(() -> getInventory().getSelectedItemName() != null && getInventory().getSelectedItemName().equals("Grain"), 5000);
+						
+					}
+					
+				} else {
+					getWalking().webWalk(upper);
+				}
+				
+			}
+			
+			// Operate machine
+			if(!operated && put) {
+				
+				if(upper.contains(myPlayer())){
+					
+					RS2Object controls = objects.closest(n -> n.getName().equals("Hopper controls"));
+
+					if(controls != null && controls.interact("Operate"))
+						Sleep.sleepUntil(() -> !myPlayer().isMoving() && getChatbox().contains(MessageType.GAME, "You operate the hopper. The grain slides down the chute."), 10000);
+					if(getChatbox().contains(MessageType.GAME, "You operate the hopper. The grain slides down the chute."))
+						operated = true;
+					
+				} else {
+					getWalking().webWalk(upper);
+				}
+				
+			}
+			
+			// Get flour
+			if(operated && put) {
+
+				getItemFromObject(bin, "Pot of flour", "Flour bin", "Empty");
+				
+			}
+			
+		}
+    	
+    }
+    
+    private void getItemFromObject(Area place, String itemName, String objectName, String interaction) throws InterruptedException {
```

Comment:
This function and `getGroundItem` are very similar. You could combine these into one function by adding a boolean parameter `isGroundItem` or something along those lines